addons/uom/models/uom_uom.py

```diff
@@ -8,146 +8,33 @@
-            if reference_count == 0 and self._origin.id and self.uom_ids:
-                raise UserError(_('UoM category %s must have at least one reference unit of measure.', self.name))
-            if self.reference_uom_id:
-                new_reference = self.uom_ids.filtered(lambda o: o.uom_type == 'reference' and o._origin.id != self.reference_uom_id.id)
-            else:
-                new_reference = self.uom_ids.filtered(lambda o: o.uom_type == 'reference' and o._origin.uom_type != 'reference')
-            if new_reference:
-                other_uoms = self.uom_ids.filtered(lambda u: u._origin.id) - new_reference
-                for uom in other_uoms:
-                    uom.factor = uom._origin.factor / (new_reference._origin.factor or 1)
-                    if uom.factor > 1:
-                        uom.uom_type = 'smaller'
-                    else:
-                        uom.uom_type = 'bigger'
-                self.reference_uom_id = new_reference._origin.id
-
-
 class UomUom(models.Model):
     _name = 'uom.uom'
     _description = 'Product Unit of Measure'
+    _parent_name = 'relative_uom_id'
     _order = "factor DESC, id"
 
     def _unprotected_uom_xml_ids(self):
         return [
-            "product_uom_hour", # NOTE: this uom is protected when hr_timesheet is installed.
-            "product_uom_dozen",
+            "product_uom_hour",  # NOTE: this uom is protected when hr_timesheet is installed.
+            "product_uom_pack_of_6",
         ]
 
-    name = fields.Char('Unit of Measure', required=True, translate=True)
-    category_id = fields.Many2one(
-        'uom.category', 'Category', required=True, ondelete='restrict',
-        help="Conversion between Units of Measure can only occur if they belong to the same category. The conversion will be made based on the ratios.")
-    factor = fields.Float(
-        'Ratio', default=1.0, digits=0, required=True,  # force NUMERIC with unlimited precision
-        help='How much bigger or smaller this unit is compared to the reference Unit of Measure for this category: 1 * (reference unit) = ratio * (this unit)')
-    factor_inv = fields.Float(
-        'Bigger Ratio', compute='_compute_factor_inv', digits=0,  # force NUMERIC with unlimited precision
-        readonly=True, required=True,
-        help='How many times this Unit of Measure is bigger than the reference Unit of Measure in this category: 1 * (this unit) = ratio * (reference unit)')
-    rounding = fields.Float(
-        'Rounding Precision', default=0.01, digits=0, required=True,
-        help="The computed quantity will be a multiple of this value. "
-             "Use 1.0 for a Unit of Measure that cannot be further split, such as a piece.")
+    name = fields.Char('Unit', required=True, translate=True)
+    relative_factor = fields.Float(
+        'Contains', default=1.0, digits='Product Unit of Measure', required=True,  # force NUMERIC with unlimited precision
```

Comment:
Should be `digits=0`, otherwise it **will** mess with non-trivial conversions

Reply:
But in that case, how can we handle cases such as imperial units, foot => meter, pound => kilogram, etc.. Those conversion ratios cannot be defined in integers, or am I missing something?

Reply:
`digits=0` means unlimited digits, not "no digits", as the comment suggests. :wink:

Reply:
Ah, okay. I indeed thought it meant integers :D